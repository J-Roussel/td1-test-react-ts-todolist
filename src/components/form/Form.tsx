import React, {FC} from "react";
import './form.css';
import { IForm } from "../../utils/interfaces";


const Form:FC<IForm> = ({valueName, valueDescri, initialValue, handleChange, addTask}):JSX.Element => {

    
    return (
        <div className='form-container'>
            <div className="form">
                <div className="info">
                    <div className="name" >
                        <label className="label-name" htmlFor="name">&nbsp; &nbsp; &nbsp;Name</label>
                        <input className="input-name" type="text" value={valueName}  onChange={handleChange} name='inputName' placeholder={initialValue[0]}/>
                    </div>
                    <div className="description">
                        <label className="label-description" htmlFor="description">&nbsp; &nbsp; &nbsp;Description</label>
                        <input className="input-description" type="textarea" value={valueDescri} onChange={handleChange} name='inputDescri' placeholder={initialValue[1]}/>
                    </div>
                </div>
                <input type="button" className="valider" value="ADD ToDo" onClick={addTask}/>
            </div>        
        </div>
    );
}


export default Form;