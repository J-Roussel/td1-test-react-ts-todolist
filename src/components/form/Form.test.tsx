import React from "react";
import renderer from "react-test-renderer";
import Form from "./Form";

describe('Unit test for Form component', () => {
    test('render form component correctly', () => {
        const FORMCOMPONENT = renderer.create(<Form valueName="test" 
                                                    valueDescri="teste" 
                                                    initialValue={["test1", "test2"]} 
                                                    handleChange={jest.fn()} 
                                                    addTask={jest.fn()}
                                              />);
        let tree = FORMCOMPONENT.toJSON();
        expect(tree).toMatchSnapshot();

    });

    test('test input', () => {
      const handleChange = jest.fn();
      const FORMCOMPONENT = renderer.create(<Form valueName="test" 
                                                  valueDescri="teste" 
                                                  initialValue={["test1", "test2"]} 
                                                  handleChange={handleChange} 
                                                  addTask={jest.fn()}
                                            />);
      let instance = FORMCOMPONENT.root;
      let tree = instance.findAllByType('input')[0];
      tree.props.onChange({ target: { name: 'inputName', value: 'Test task' } });
      expect(handleChange).toHaveBeenCalledWith({ target: { name: 'inputName', value: 'Test task' } });

    })

    
 



})

