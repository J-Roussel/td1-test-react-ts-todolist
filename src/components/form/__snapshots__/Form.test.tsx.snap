// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for Form component render form component correctly 1`] = `
<div
  className="form-container"
>
  <div
    className="form"
  >
    <div
      className="info"
    >
      <div
        className="name"
      >
        <label
          className="label-name"
          htmlFor="name"
        >
               Name
        </label>
        <input
          className="input-name"
          name="inputName"
          onChange={[MockFunction]}
          placeholder="test1"
          type="text"
          value="test"
        />
      </div>
      <div
        className="description"
      >
        <label
          className="label-description"
          htmlFor="description"
        >
               Description
        </label>
        <input
          className="input-description"
          name="inputDescri"
          onChange={[MockFunction]}
          placeholder="test2"
          type="textarea"
          value="teste"
        />
      </div>
    </div>
    <input
      className="valider"
      onClick={[MockFunction]}
      type="button"
      value="ADD ToDo"
    />
  </div>
</div>
`;

exports[`Unit test for Form component renders form component correctly 1`] = `null`;
